### ISSUE TITLE
Rename the types and interfaces to do with the defences

### ISSUE THREAD

**Issue by pmarsh-scottlogic** [gsproston-scottlogic/prompt-injection#462]
I would like to change the naming of some things to make my life easier.

# What we have at the moment:

- `Defences` is a variable in the level state that contains all the relevant information on the setup of the defence for a particular level. It is in array of `DefenceInfo`s.
- `DefenceInfo` contains the information for one specific defence type (`CHARACTER_LIMIT`, `EMAIL_WHITELIST`, etc). It has a config property which obviously specifies how it is configured. Now, this is an array because some defence types have multiple configurable options, e.g.  `EVALUATION_LLM_INSTRUCTIONS` has a configurable pre-prompt for each of the Prompt injection detection and malicious input detection LLMs)
- `DefenceConfig` contains information about how one specific option is configured.

To sum up
```
defences: DefenceInfo[]

class DefenceInfo {
  id: DEFENCE_TYPES;
  config: DefenceConfig[];
  isActive: boolean;
  isTriggered: boolean;
}

interface DefenceConfig {
  id: string;
  value: string;
}
```

# What I would like:

I want to change
- `DefenceInfo` -> `Defence`
- `DEFENCE_TYPES` -> `DEFENCE_ID`
- `DefenceConfig` -> `DefenceConfigItem`

I think this is more descriptive of the structure of things.

```
defences: Defence[]

class Defence {
  id: DEFENCE_ID;
  config: DefenceConfigItem[];
  isActive: boolean;
  isTriggered: boolean;
}

interface DefenceConfigItem {
  id: string;
  value: string;
}
```

what do we think?
